parcels/interaction/baseinteractionkernel.py

```diff
@@ -0,0 +1,145 @@
+import re
+import inspect
+from sys import version_info
+
+try:
+    from mpi4py import MPI
+except:
+    MPI = None
+
+from parcels.kernel.basekernel import BaseKernel
+
+__all__ = ['BaseKernel']
+
+
+re_indent = re.compile(r"^(\s+)")
+
+
+class BaseInteractionKernel(BaseKernel):
```

Comment:
after reviewing the class, I see you add another in-between class layer here to facilitate a more slim-line `InteractionKernel` specialisation class - good idea.